File: num.cpp
#include <iostream>

std::string numberToWords(int num) {
    if (num == 0) return "zero";

    std::string belowTwenty[] = {"", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"};
    std::string tens[] = {"", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"};
    std::string thousands[] = {"", "thousand", "million", "billion"};

    std::string result = "";
    int i = 0;

    while (num > 0) {
        if (num % 1000 != 0) {
            std::string chunk = "";
            int n = num % 1000;

            if (n >= 100) {
                chunk += belowTwenty[n / 100] + " hundred ";
                n %= 100;
            }
            if (n >= 20) {
                chunk += tens[n / 10] + " ";
                n %= 10;
            }
            if (n > 0) {
                chunk += belowTwenty[n] + " ";
            }

            result = chunk + thousands[i] + " " + result;
        }
        num /= 1000;
        i++;
    }

    return result;
}

int romanToInt(const std::string& s) {
    int total = 0;
    int prevValue = 0;

    for (int i = s.size() - 1; i >= 0; --i) {
        int value = 0;
        switch (s[i]) {
            case 'I': value = 1; break;
            case 'V': value = 5; break;
            case 'X': value = 10; break;
            case 'L': value = 50; break;
            case 'C': value = 100; break;
            case 'D': value = 500; break;
            case 'M': value = 1000; break;
            default: return -1; // Invalid character
        }

        if (value < prevValue) {
            total -= value;
        } else {
            total += value;
        }
        prevValue = value;
    }

    return total;
}

int main() {
    std::string roman;
    std::cout << "Enter Roman numeral: ";
    std::cin >> roman;

    int result = romanToInt(roman);
    if (result == -1) {
        std::cout << "Invalid Roman numeral!" << std::endl;
    } else {
        std::cout << "Integer: " << result << std::endl;
        std::cout << "Word: " << numberToWords(result) << std::endl;
    }
    return 0;
}

File: num.py
def number_to_words(num):
    if num == 0:
        return "zero"
    
    below_twenty = ["", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten", 
                    "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"]
    tens = ["", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"]
    thousands = ["", "thousand", "million", "billion"]
    
    result = ""
    i = 0
    
    while num > 0:
        if num % 1000 != 0:
            chunk = ""
            n = num % 1000
            
            if n >= 100:
                chunk += below_twenty[n // 100] + " hundred "
                n %= 100
                
            if n >= 20:
                chunk += tens[n // 10] + " "
                n %= 10
                
            if n > 0:
                chunk += below_twenty[n] + " "
                
            result = chunk + thousands[i] + " " + result
            
        num //= 1000
        i += 1
        
    return result.strip()

def roman_to_int(s):
    total = 0
    prev_value = 0
    
    roman_values = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    }
    
    for char in reversed(s):
        if char not in roman_values:
            return -1
            
        value = roman_values[char]
        if value < prev_value:
            total -= value
        else:
            total += value
        prev_value = value
        
    return total

if __name__ == "__main__":
    roman = input("Enter Roman numeral: ")
    result = roman_to_int(roman)
    
    if result == -1:
        print("Invalid Roman numeral!")
    else:
        print(f"Integer: {result}")
        print(f"Word: {number_to_words(result)}")